MapboxCoreNavigation/RouteController.swift

```diff
@@ -330,6 +339,50 @@ extension RouteController: CLLocationManagerDelegate {
         }
     }
     
+    func checkForFasterRoute(from location: CLLocation) {
+        guard let currentUpcomingManeuver = routeProgress.currentLegProgress.upComingStep else { return }
+        
+        guard let lastLocationDate = lastLocationDate else {
+            self.lastLocationDate = location.timestamp
+            return
+        }
+
+        // Only check ever 2 minutes for faster route
+        guard location.timestamp.timeIntervalSince(lastLocationDate) >= 120 else { return }
+        let durationRemaining = routeProgress.durationRemaining
+        let currentAlertLevel = routeProgress.currentLegProgress.alertUserLevel
+        
+        getDirections(from: location) { [weak self] (route, error) in
+            guard let strongSelf = self else { return }
+            
+            strongSelf.lastLocationDate = nil
+            
+            guard let route = route else { return }
+            
+            if route.legs[0].steps[1].expectedTravelTime <= 70 {
+                let maneuverLocation = CLLocation(coordinate: currentUpcomingManeuver.maneuverLocation, altitude: 0, horizontalAccuracy: 1, verticalAccuracy: 1, course: currentUpcomingManeuver.finalHeading ?? currentUpcomingManeuver.initialHeading ?? location.course, speed: 10, timestamp: Date())
+                
+                strongSelf.getDirections(from: maneuverLocation, completion: { (route, error) in
+                    guard let route = route else { return }
+                    
+                    // Only use new route if it's at least 10% faster
+                    if route.expectedTravelTime <= 0.9 * durationRemaining {
+                        // If the upcoming maneuver in the new route is the same as the current upcoming maneuver, don't announce it
+                        strongSelf.routeProgress = RouteProgress(route: route, legIndex: 0, alertLevel: currentUpcomingManeuver.description == route.legs[0].steps[1].description ? currentAlertLevel : .none)
+                        strongSelf.delegate?.routeController?(strongSelf, didRerouteAlong: route)
+                    }
+                })
+            } else {
+                // Only use new route if it's at least 10% faster
+                if route.expectedTravelTime <= 0.9 * durationRemaining {
```

Comment:
There’s a fair amount of duplication within this method. This `if` statement represents the case where the new route leaves enough time to maneuver. Since it appears twice in the same method, let’s factor it out into a local closure, then call it in both places.